src/Microsoft.ML.AutoML/AutoMLExperiment/AutoMLExperiment.cs

```diff
@@ -0,0 +1,262 @@
+        {
+            this._settings.DatasetSettings = new TrainTestDatasetSettings()
+            {
+                TrainDataset = train,
+                TestDataset = test
+            };
+
+            return this;
+        }
+
+        public AutoMLExperiment SetDataset(IDataView dataset, int fold = 10)
+        {
+            this._settings.DatasetSettings = new CrossValidateDatasetSettings()
+            {
+                Dataset = dataset,
+                Fold = fold,
+            };
+
+            return this;
+        }
+
+        public AutoMLExperiment SetTunerFactory<TTunerFactory>()
+            where TTunerFactory : ITunerFactory
+        {
+            var descriptor = new ServiceDescriptor(typeof(ITunerFactory), typeof(TTunerFactory), ServiceLifetime.Singleton);
+            if (this._serviceCollection.Contains(descriptor))
+            {
+                this._serviceCollection.Replace(descriptor);
+            }
+            else
+            {
+                this._serviceCollection.Add(descriptor);
+            }
+
+            return this;
+        }
+
+        public AutoMLExperiment SetMonitor(IMonitor monitor)
+        {
+            var descriptor = new ServiceDescriptor(typeof(IMonitor), monitor);
+            if (this._serviceCollection.Contains(descriptor))
+            {
+                this._serviceCollection.Replace(descriptor);
+            }
+            else
+            {
+                this._serviceCollection.Add(descriptor);
+            }
+
+            return this;
+        }
+
+        public AutoMLExperiment SetPipeline(MultiModelPipeline pipeline)
+        {
+            this._settings.Pipeline = pipeline;
+            return this;
+        }
+
+        public AutoMLExperiment SetTrialRunnerFactory(ITrialRunnerFactory factory)
+        {
+            var descriptor = new ServiceDescriptor(typeof(ITrialRunnerFactory), factory);
+            if (this._serviceCollection.Contains(descriptor))
+            {
+                this._serviceCollection.Replace(descriptor);
+            }
+            else
+            {
+                this._serviceCollection.Add(descriptor);
+            }
+
+            return this;
+        }
+
+        public AutoMLExperiment SetPipeline(SweepableEstimatorPipeline pipeline)
+        {
+            var res = new MultiModelPipeline();
+            foreach (var e in pipeline.Estimators)
+            {
+                res = res.Append(e);
+            }
+
+            this.SetPipeline(res);
+
+            return this;
+        }
+
+        public AutoMLExperiment SetEvaluateMetric(BinaryClassificationMetric metric, string predictedColumn = "Predicted", string truthColumn = "label")
+        {
+            this._settings.EvaluateMetric = new BinaryMetricSettings()
+            {
+                Metric = metric,
+                PredictedColumn = predictedColumn,
+                TruthColumn = truthColumn,
```

Comment:
I must have missed this in the prior PR's, but it would be better to label this as `LabelColumn` instead of `TruthColumn`.